### PR TITLE
Clone picker rows from an inert template prototype; harden JSON edge validation

### PR DESCRIPTION
**Pull request by dkastl**
## The bug (follow-up to #104)

Row deletion works since #104 — but delete **every** row and the add link dies: it cloned the *last existing row*, and with zero rows there was nothing to clone. Second failure of the clone-a-sibling approach (the first was "Add attachment" with zero initial rows), so this removes the pattern instead of patching it again:

- each picker now renders an inert **`<template>` prototype**; the add link clones that, fully independent of the visible rows
- the row markup moved into `_entity_row` / `_attachment_row` partials, shared by stored rows and prototypes — one source of truth
- side effect: new rows arrive prefilled like the starter row (`.*` pattern) instead of as a cleared clone

Browser-verified: remove all → add works, both pickers.

## Edge-case hardening + tests

The form's serialization contract had validation gaps: `take_json_entities` / `take_json_attachments` only rejected *unparsable* JSON, so parsable-but-wrong shapes (`{}`, `[1, 2]`, `"string"`) passed validation and would break the subscription builders or the picker downstream.

- `entities_string` must now be a **non-empty array of objects**
- `attachments_string` must be an **array of objects**; `null` (the picker's no-rows value) still clears
- `geometry_string` deliberately stays free-form (`"${location}"` templates, GeoJSON objects, `null` are all legitimate)

New unit tests pin every accepted/rejected shape for `entities_string`, `attachments_string`, `geometry_string` and `attrs`, plus the `null`-clears-attachments round trip. Functional tests pin the prototype contract (one row inside each `<template>`), so a picker that loses its prototype fails CI instead of failing users.

## Testing

Full suite: **168 runs, 617 assertions, 0 failures** (was 161 — 7 new tests).

### CODE REVIEW

File: app/models/subscription_template.rb [hunk 1]
```diff
@@ -160,8 +160,16 @@ def set_default_notify_on_metadata_change
...
 
+  # Parsable JSON is not enough: the subscription builders and the form's
+  # structured picker both expect an array of entity objects, so '{}' or
+  # '[1]' must fail validation here instead of breaking publish later.
   def take_json_entities
-    self.entities = JSON.parse(entities_string)
+    parsed = JSON.parse(entities_string)
+    unless parsed.is_a?(Array) && parsed.any? && parsed.all? { |e| e.is_a?(Hash) }
+      errors.add :entities_string, I18n.t('model.subscription_template.must_be_valid_array_of_objects')
+      return
```
[the remark this answers is not in the document]
Comment (dkastl) [reply app/models/subscription_template.rb:175]: Fixed in 0a1f3dd — `take_json_entities` now returns early on blank (the presence validation reports it, exactly one error, no `TypeError` from `JSON.parse(nil)`), and both validators parse `…_string.to_s` so a stray non-string writer value lands on the graceful rescue path instead of raising. Pinned by `test_blank_entities_string_reports_presence_only`.
